src/components/protectedPage.jsx

```diff
@@ -0,0 +1,27 @@
+import { authenticate } from "@/app/utils/authenticate";
+
+const ProtectedPage = async ({
+  fallback,
+  unauthenticatedComponent,
+  children,
+}) => {
+  const authResult = await authenticate();
+
+  if (authResult === null) {
+    return fallback ? fallback : <></>;
+  } else if (authResult.auth === 200) {
+    return children;
+  } else {
+    if (unauthenticatedComponent) {
+      return <>{unauthenticatedComponent}</>;
```

Comment:
this will leaving loading on the page for a second cause authenticate is await, I would opt for not putting protected pages on pages that dont need it instead of bypassing